docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-web-data-model/custom-sessionization-and-users/index.md

```diff
@@ -0,0 +1,61 @@
+
+Traditionally a session in this sense has been a _true_ session i.e. it has been the `domain_sessionid` value that has identified which events belong to the same session, however over time this became a limitation of the package as users wanted to define their session using a different field or logic, which meant they were unable to use our package and take advantage of the complex ground work we provided. Since version 0.16.0 of the web package, it is possible to overwrite the `domain_sessionid` field as part of the package logic and use your own session identifier, to ensure all events within a "session" are processed in the same run, whatever a "session" in this case means to you.
+
+:::caution
+
+To reduce the number of breaking changes within the web package, we made the decision to have the custom session identifier to overwrite the value of the `domain_sessionid` field in the `snowplow_web_base_events_this_run` and all downstream tables, but have added an `original_domain_sessionid` field to these tables to keep the original value if needed.
+
+:::
+
+## Examples
+
+A deeper set of documentation is available in the [Advanced Usage of Snowplow Utils](/docs/modeling-your-data/modeling-your-data-with-dbt/dbt-models/dbt-utils-data-model/dbt-utils-advanced-operation/index.md?warehouse=redshift%2Bpostgres#utilizing-custom-contexts-or-sdes) page, including how to provide a hierarchy of fields in case one is null, but here are two short example showcasing how you can use a field in a custom context, and how you can provide custom SQL to create your "session" identifier.
+
+:::tip
+
+Remember that any events with a null "session" identifier will be excluded from the `snowplow_base_events_this_run` table and any processing of the package, make sure your identifier has a value for all events you want to process!
+
+:::
+
+
+
+#### Custom Context
+
+This example uses a field called `session_id` in your `com_mycompany_session_identifier_1` context and will use this as the value in your `domain_sessionid` field for all tables (excluding the manifest tables, which uses a column named `session_identifier`).
+
+```yml title="dbt_project.yml"
+vars:
+    snowplow_web:
+        snowplow__session_identifiers: [{'schema': 'com_mycompany_session_identifier_1', 'field': 'session_id', 'prefix': 'si'}]
```

Comment:
This description might be confusing as depending on the data warehouse this might not be as simple as this. E.g. for Snowflake they would have to use the camel cased field (are users aware of this?) and for bigquery currently it does not support multiple schema versions. Are we planning to support that soon or is that the idea that they would need to use custom sql for that with the combine column versions macro in it?

Reply:
Sorry, I forgot to add automatically so that they don't have to touch the model upon schema change. But just thinking as it is a variable they can just change it at runtime so it can actually work hassle free then!

Reply:
PR in for this now :D